File: circular-array-rotation/circular-array-rotation.php
<?php

// Complete the circularArrayRotation function below.
function circularArrayRotation($a, $k, $queries, $n) {
    foreach ($queries as $i) {
        print $a[($i + $n - $k) % $n] . "\n";
    }
}

$line = trim(fgets(STDIN));
list($n, $k, $q) = explode(' ', $line);

$line = trim(fgets(STDIN));
$a = array_map('intval', explode(' ', $line));

$queries = [];

for ($i = 0; $i < $q; $i++) {
    fscanf(STDIN, "%d\n", $queries_item);
    $queries[] = $queries_item;
}

circularArrayRotation($a, $k, $queries, $n);

File: circular-array-rotation/circular-array-rotation.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the circularArrayRotation function below.
def circularArrayRotation(a, k, queries, n):
    output = []
    for i in queries:
        output.append(a[(i + n - k) % n])

    return output


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    nkq = input().split()

    n = int(nkq[0])

    k = int(nkq[1])

    q = int(nkq[2])

    a = list(map(int, input().rstrip().split()))

    queries = []

    for _ in range(q):
        queries_item = int(input())
        queries.append(queries_item)

    result = circularArrayRotation(a, k, queries, n)

    fptr.write('\n'.join(map(str, result)))
    fptr.write('\n')

    fptr.close()
